viewing_party/party.py

```diff
@@ -1,23 +1,183 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+
+    new_movie = {}
+
+    # check that all necessary movie characteristics exist before creating new movie
+    if title and genre and rating:
+        new_movie["title"] = title
+        new_movie["genre"] = genre
+        new_movie["rating"] = rating
+    else:
+        new_movie = None
+
+    return new_movie
+
+def add_to_watched(user_data, movie):
+
+    # add movie to watched list
+    user_data["watched"].append(movie)
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+
+    # add movie to watchlist
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+def watch_movie(user_data, movie_title):
+
+    # move movies that user watches from watchlist to watched
+    for movie in user_data["watchlist"]:
+        if movie["title"] == movie_title:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+    
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+
+    # variable to hold sum of all movie ratings
+    total = 0
+
+    for movie in user_data["watched"]:
+        # add each movie's rating to the total
+        total += movie["rating"]
+    
+    # calculate average rating for all movies within "watched"
+    if total == 0:
+        rating_average = 0
+    else:
+        rating_average = total/len(user_data["watched"])
+
+    return rating_average
+
+def get_most_watched_genre(user_data):
+    
+    # empty dictionary to hold movie genres and their frequency
+    movie_genre_count = {}

```

Comment:
I suggest moving the `movie_genre_count` declaration inside the else-block so that we don't allocate space for the variable unless we know it will be used.